### DISCUSSION TITLE
Status of the project

### DISCUSSION THREAD

**Discussion by ffenix113** [ffenix113/zigbee_home#66] [General]
Hello everyone, 👋 

It seems that this project is of interest to quite some people, and as it is gaining commits so slowly, I wanted to share an update.

Unfortunately, as all of you might have seen, the development is not as rapid as it can be.
The reason is that last couple of month and also some month into the future, I am quite busy with other things, which leaves me little time for my hobbies, including this project.

But this still does not mean that this project is not being worked on. Plans for this project are only growing, and with the help of community members that donated Development Board, opened issues & discussions, and even support me on BuyMeACoffe(!) (thank you, @Hedda, @rsporsche, @bbadrignans and others), I can work on improving the stability and required functionality to improve usability and usefulness of this project. Even for this young project, people help as much as they can, and I cannot even start to express how I am grateful for this...

More so, before I become as busy as I am now, I was working on board specifically for this project that can be shared with the community and everybody else. The components already arrived for a couple of prototype boards, and the PCBs should arrive in a week or two. This will also be something that will help the project to move forward.

### Bottom line is: the project is still active 🚀, even if development has slowed down.

I have no plans to abandon it, only to make it better!

Thank you.

**Comment by ffenix113** [comment]
The first custom board for this project is alive, and can even blink LEDs! Based on nRF5340 and Arduino Nano board dimensions.
![20240714_165800_1](https://github.com/user-attachments/assets/af4395c6-95a5-4ffb-ae07-bde2c6e1e2d2)

**Reply by Hedda** [reply]
Nice! 

Does it have a USB-to-UART converter chip (a.k.a USB-to-Serial bridge chip) to enable communication and progamming/flashing over USB?

Do you plan on release/publishing schematics and under a open source hardware license?

https://en.wikipedia.org/wiki/Open-source_hardware

**Reply by ffenix113** [reply]
Hello Hedda,

> Does it have a USB-to-UART converter chip (a.k.a USB-to-Serial bridge chip) to enable communication and progamming/flashing over USB?

I don't think that nRF chips need it. They can be programmed via USB with bootloaders like MCUBoot and [Adafruit's bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader). But please correct me if I am wrong. 

P.S.: The last 2 bottom pins on the right(hugged by a white line) are SWD and SWCLK, which might help as well.

> Do you plan on release/publishing schematics and under a open source hardware license?

Absolutely! I have not yet licensing requirements for it, but I want to provide complete design available in this, or related, repository.

**Reply by Hedda** [reply]
> > Does it have a USB-to-UART converter chip (a.k.a USB-to-Serial bridge chip) to enable communication and progamming/flashing over USB?
> 
> I don't think that nRF chips need it. They can be programmed via USB with bootloaders like MCUBoot and [Adafruit's bootloader](https://github.com/adafruit/Adafruit_nRF52_Bootloader). But please correct me if I am wrong.

Not sure myself, but bonus if the board could optionally be re-purposed with ZBOSS NCP firmware instead to be used as a Zigbee Coordinator for Home Assistant’s ZHA integration via the zigpy-zbozz project and for that you need a USB-to-Serial interface:

https://community.home-assistant.io/t/nordic-semi-nrf52840-chips-with-zboss-zigbee-stack-now-usable-as-zigbee-coordinator-for-the-zha-integration-via-new-radio-library-for-zigpy/570196/

https://github.com/kardia-as/zigpy-zboss

https://github.com/kardia-as/nrf-zboss-ncp

https://www.home-assistant.io/integrations/zha
